File: higher-order-binary-ops/less-than-100.scala
def lt(ro: Int)(lo: Int): Boolean = lo < ro

val num = List(5, 500, 50, 101, 200, 35)
val out = num filter { lt(100) } // or: num filter { _ < 100 } since the < operator is a curried function.

println(s"less than 100: [ ${out.mkString(", ")} ]")
//=> less than 100: [ 5, 50, 35 ]

File: higher-order-binary-ops/less-than-100.swift
func lt(ro: Int)(lo: Int) -> Bool {
  return lo < ro
}

let num = [5, 500, 50, 101, 200, 35]
let out = num.filter(lt(100))

println("less than 100: \(out)")
//=> less than 100: [5, 50, 35]
